src/lava/magma/runtime/message_infrastructure/factory.py

```diff
@@ -5,6 +5,8 @@
 from lava.magma.runtime.message_infrastructure.multiprocessing import \
     MultiProcessing
 
+"""Factory class to create the messaging infrastructure"""
```

Comment:
I am using this example for the general question of documenting within py files. I think we do not have this file based comments anywhere else. 
Should we do that for every file going forward?

Reply:
We used to have this in nxsdk. Just picked a random file from Tensorflow - https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/compiler/xla/jit.py

I think it gives a context to the reader as a file might have multiple classes and but typically a single goal and responsibility and all classes defined inside are somehow related with each other loosely.